### PR TITLE
Fix #829 : Report an issue switches to normal tab even with PB only e…

### LINKED ISSUE #829: Report an issue switches to normal tab even with PB only enabled
### Description: 
Report an issue switches to normal tab even with PB only enabled. Issue originally reported on community https://community.brave.com/t/report-a-bug-forces-user-out-of-private-browsing/43681

### Steps to Reproduce 
  1. Enable PB mode only from settings
  2. Open settings and go to Report an issue
  3. Normal tab is opened, Can't open tab container have to exit PB mode to actually see the normal tab where the community page is opened

**Actual result:** 
Report an issue always switches to normal tab even with PB only enabled

**Expected result:**
If report an issue is triggered via PB, then it should be opened in a PT only

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
All

**Device details:** 
All

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by gheilig**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adaquate test plan exists for QA to validate (if applicable)

### CODE REVIEW

**Review by kylehickinson — APPROVED** [review]

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1361,7 +1361,8 @@ extension BrowserViewController: QRCodeViewControllerDelegate {
 
 extension BrowserViewController: SettingsDelegate {
     func settingsOpenURLInNewTab(_ url: URL) {
-        self.openURLInNewTab(url, isPrivileged: false)
+        let forcedPrivate = Preferences.Privacy.privateBrowsingOnly.value || PrivateBrowsingManager.shared.isPrivateBrowsing
```
Comment (jhreis) [inline Client/Frontend/Browser/BrowserViewController.swift]: `PrivateBrowsingManager.shared.isPrivateBrowsing` should be good enough here, as this is set if `privateBrowsingOnly` is `true`. Were you seeing different behavior?

**Review by jhreis — CHANGES_REQUESTED** [review]
Thanks bunches for the PR! Really love the fix.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (gheilig) [reply Client/Frontend/Browser/BrowserViewController.swift]: You are right. Thank you. fixed. (btw copy & paste from `NavigationRouter`)

**Review by jhreis — APPROVED** [review]

**Comment by jhreis** [comment]
Really appreciate the work on this, nice fix!

**Pull request merged**